src/domain/common/profile/Profile.ts

```diff
@@ -19,3 +19,11 @@ export interface Reference {
   uri: string;
   description?: string;
 }
+
+export interface ReferenceWithAuthorization {
+  id: string;
+  name: string;
+  uri: string;
+  description?: string;
+  authorization?: Partial<Authorization>;
```

Comment:
I guess you're only using this to check myPrivileges, can you change this to: 
```
authorization?: {
   myPrivileges: AuthorizationPrivilege[];
}
``` 
or something similar?